grails-app/services/uk/ac/ox/softeng/maurodatamapper/plugins/csv/datamodel/provider/importer/CsvDataModelImporterProviderService.groovy

```diff
@@ -72,73 +79,158 @@ class CsvDataModelImporterProviderService
     }
 
     DataModel importSingleFile(CsvDataModelImporterProviderServiceParameters parameters) {
-        Map<String, ColumnData> columns = getColumnDefinitions(parameters)
-
-        if (!columns) {
-            throw new ApiBadRequestException('CSV02', "Nothing to load into DataModel from file [${parameters.importFile.fileName}]")
-        }
+        log.info('Loading CSV model from {}', parameters.getImportFile().getFileName())
+        String fileType = parameters.getImportFile().getFileType()
+        log.info('Loading CSV model filetype {}', fileType)
 
         String modelName = parameters.importFile.fileName.take(parameters.importFile.fileName.lastIndexOf('.'))
         DataModel dataModel = new DataModel(label: modelName, modelType: DataModelType.DATA_ASSET, authority: authorityService.defaultAuthority)
 
         Map<String, DataType> dataTypes = getPrimitiveDataTypes(parameters)
-        dataTypes.each { k, dt ->
+        dataTypes.each {k, dt ->
             dataModel.addToDataTypes(dt)
         }
 
-        DataClass topLevelClass = new DataClass(label: 'CSV fields')
-        dataModel.addToDataClasses(topLevelClass)
+        Boolean importFromArchive = false
+
+        List<CsvDataModelImporterProviderServiceParameters> parametersList = []
+
+        if (fileType == "application/zip" || fileType == "application/x-zip-compressed") {
+            importFromArchive = true
+            log.info('Loading Zip File')
+
+            if (!parameters.firstRowIsHeader) {
+                parameters.firstRowIsHeader = true
+                log.warn('Using first row of CSV files as headers when importing from archive')
+            }
+
+            File tempDir = Files.createTempDirectory("temp").toFile()
+            log.info('Temp Folder Location {}', tempDir.getAbsolutePath())
 
-        columns.each { header, column ->
-            DataType dataType = column.getDataType(dataTypes)
-            if (dataType instanceof EnumerationType) {
-                dataModel.addToDataTypes(dataType)
+            List<Pattern> excludePatterns = grailsApplication.config.getProperty('maurodatamapper.csv.archive.exclude', List).collect {
```

Comment:
i wouldnt use exclude patterns, you are ONLY interested in csv files, so ONLY extract files with .csv

Reply:
Have removed exclude patterns and extracted all .csv files